lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/LibertyUtils.java

```diff
@@ -253,19 +253,24 @@ public static String getVersion(String serverXMLUri) {
      * Return temp directory to store generated feature lists and schema. Creates
      * temp directory if it does not exist.
      * 
-     * @param folder WorkspaceFolderURI indicates where to create the temporary
-     *               directory
+     * @param libertyWorkspace LibertyWorkspace where the temporary directory is made
+     *               
      * @return temporary directory File object
      */
-    public static File getTempDir(String workspaceFolderURI) {
-        if (workspaceFolderURI == null) {
+    public static File getTempDir(LibertyWorkspace libertyWorkspace) {
+        if (libertyWorkspace == null) {
             return null;
         }
         try {
-            URI rootURI = new URI(workspaceFolderURI);
-            Path rootPath = Paths.get(rootURI);
-            File file = rootPath.toFile();
-            File libertyLSFolder = new File(file, ".libertyls");
+            Path targetFolderPath = findFileInWorkspace(libertyWorkspace, Paths.get("target"));
+            Path buildFolderPath  = findFileInWorkspace(libertyWorkspace, Paths.get("build"));
```

Comment:
This seems too generic and error prone to me. It would match any folder ending with `target` or `build` wouldn't it? So if a workspace had a folder name `dev-target` it would match? Or `testbuild`?

Reply:
How do we find the installed Liberty to use the schema gen jar?

Reply:
What if we looked for the `liberty-plugin-config.xml` file and used the parent dir of that? I have a feeling we will want to use values from that file down the road anyhow.

Reply:
Yeah, I think that makes more sense. I'll update it to look for that file.